### PR TITLE
Ds 6522 cluster selection

### PR DESCRIPTION
**Pull request by ilsep93**
#### *__Why__ is this code necessary?*
First step in selecting ZCTAs in the US. Future notebook in another branch will use these functions to select a list of ZCTAs in the US based on a design effect, budget, and previous estimate (2022 NFID study).

#### *__What__ is the code achieving?* 
Develops business logic for computing required sample size and selecting clusters (eg. zip codes) from a sampling frame (eg. list of all zip codes or ZCTAs in the US)

#### *__What__ specifically are you looking to get out of the review?*
- [X] Correctness
- [X] Readability
- [X] Organization of Code
- [ ] Solution Approach
- [X] Increase Knowledge Transfer
- [ ] Other (Please detail below)

#### *__How__ do you suggest the reviewer review it? What specific steps should they take for a successful review from them and a useful review for you?*
From the root of the repo, type in your terminal:
```bash
poetry shell
poetry install
pytest
```
You might need to open and close the poetry shell once in order for pytest to be recognized as an import. 

All tests should pass.

## Developer Checklist
- [X] I've added documentation where possible
- [X] I've run pre-commit hooks
- [X] I've added unit tests where possible
- [ ] I've added inline Github comments to point out notable code changes or justifications for decisions

## Reviewer Checklist
- [ ] I've run unit tests locally
- [ ] I'm able to set up the environment
- [ ] I understand what the code is trying to accomplish

### CODE REVIEW

**Review by sednaasil — DISMISSED** [review]
LGTM.  All tests have passed :)

**Comment by sednaasil** [comment]
To run the poetry shell and install I had to remake the toml file.  I just pushed it your way.

**Review by sednaasil — APPROVED** [review]
pending changes to *.toml

File: pyproject.toml [hunk 1]
```diff
@@ -1,17 +1,20 @@
 [tool.poetry]
-name = "sample-design"
+name = "repository-name" # Rename this to be reflective of the repository name
```
Comment (ilsep93) [inline pyproject.toml:2]: @sednaasil could we change this back to "sample-design"? It would be the name of the poetry shell

File: pyproject.toml [hunk 2]
```diff
@@ -1,17 +1,20 @@
...
 version = "0.1.0"
-description = ""
+description = "Add a descrition similar to the repo description on GitHub"
```
Comment (ilsep93) [inline pyproject.toml:4]: Maybe "Environment for sample design"

**Pull request merged**